includes/forum-profile.php

```diff
@@ -397,8 +398,18 @@ public function myProfileLink() {
                 // Get and build profile link.
                 $profileLink = $this->getProfileLink($currentUserObject);
 
-                echo '<a class="profile-link" href="'.$profileLink.'">'.__('Profile', 'asgaros-forum').'</a>';
+            } else {
+                $profileLink = 'profile link';
             }
+
+            return array(
+                'menu_class'            =>  'profile-link',
+                'menu_link_text'        =>  esc_html__('Profile', 'asgaros-forum'),
+                'menu_url'              =>  $profileLink,
+                'menu_login_status'     =>  1,
```

Comment:
Beim Login-Status muss man auch nochmal extra aufpassen. Es gibt ja Einstellungen, mit denen explizit definiert werden kann, dass (in diesem Beispiel das Profil) eine spezielle Seite für eingeloggte Benutzer ODER auch Gäste aufrufbar ist. Soweit ich es sehe, ist in diesem Fall das Profil NUR noch für eingeloggte Benutzer aufrufbar? Falls meine Vermutung stimmt, bitte prüfe dieses Problem auch nochmal entsprechend für die anderen anwendbaren Bereiche (Mitgliederliste, etc.).

Reply:
Beim Profil kann diese Konstellation ja nicht vorkommen, da es nur um das eigene Profil geht. Bei der Memberslist ist das noch wichtig, allerdings wird das beim Aufruf von `$this->functionality_enabled()` überprüft. Um den Array für die Nutzung besser zu gestalten habe ich jetzt den login status noch abgefragt. Dadurch wird nicht pauschaul 0 eingetragen, sondern immer Abhängigkeit von der gewählten Einstellung gesetzt.